tools/checked-c-convert/ProgramInfo.cpp

```diff
@@ -31,6 +31,163 @@ bool ProgramInfo::checkStructuralEquality(uint32_t V, uint32_t U) {
   return false;
 }
 
+bool ProgramInfo::link() {
+
+  // For every global symbol in all the global symbols that we have found
+  // go through and apply rules for whether they are functions or variables.
+
+  for (const auto &S : GlobalSymbols) {
+    // First, extract out the function symbols from S. 
+    std::set<GlobalFunctionSymbol*> funcs;
+    std::set<GlobalVariableSymbol*> vars;
+    for (const auto &U : S.second)
+      if (GlobalFunctionSymbol *K = dyn_cast<GlobalFunctionSymbol>(U))
+        funcs.insert(K);
+      else if (GlobalVariableSymbol *V = dyn_cast<GlobalVariableSymbol>(U))
+        vars.insert(V);
+
+    // Then, pairwise-iterate over each of the function symbols found for this 
+    // symbol. What we want to do is for a sequence of constraint variables on
+    // F1,F2, set the constraint variables for F1(v0,vi,vN) and F2(v0,vj,vN) to
+    // be equal to each other, i.e. to enter a series of constraints of the form
+    // vi == vj.
+    //
```

Comment:
Yes, they would. I think the comment doesn't communicate the intent of the code clearly. The iteration happens with `I` and `J` where `J` is always one-ahead of `I`, so I think the code is doing what you say. I'll try to come up with a better way to write the comment.